go/offline/rpc_cache.go

```diff
@@ -0,0 +1,127 @@
+package offline
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"github.com/keybase/client/go/chat/storage"
+	"github.com/keybase/client/go/encrypteddb"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"golang.org/x/net/context"
+	"sync"
+)
+
+type RPCCache struct {
+	sync.Mutex
+	edb *encrypteddb.EncryptedDB
+}
+
+func newEncryptedDB(g *libkb.GlobalContext) *encrypteddb.EncryptedDB {
+	keyFn := func(ctx context.Context) ([32]byte, error) {
+		return storage.GetSecretBoxKey(ctx, g, storage.DefaultSecretUI)
+	}
+	dbFn := func(g *libkb.GlobalContext) *libkb.JSONLocalDb {
+		return g.LocalDb
+	}
+	return encrypteddb.New(g, dbFn, keyFn)
+}
+
+func NewRPCCache(g *libkb.GlobalContext) *RPCCache {
+	return &RPCCache{
+		edb: newEncryptedDB(g),
+	}
+}
+
+func hash(rpcName string, arg interface{}) ([]byte, error) {
+	h := sha256.New()
+	h.Write([]byte(rpcName))
+	h.Write([]byte{0})
+	raw, err := libkb.MsgpackEncode(arg)
+	if err != nil {
+		return nil, err
+	}
+	h.Write(raw)
+	return h.Sum(nil), nil
+}
+
+func dbKey(rpcName string, arg interface{}) (libkb.DbKey, error) {
+	raw, err := hash(rpcName, arg)
+	if err != nil {
+		return libkb.DbKey{}, err
+	}
+	return libkb.DbKey{
+		Typ: libkb.DBOfflineRPC,
+		Key: hex.EncodeToString(raw[0:16]),
```

Comment:
Basically storing sha256 of rpc args in plain text on the hdd. That gonna be ok?

Reply:
yeah, i was hoping it was ok, and that the marshaling of msgpack is deterministic!